File: dart/lib/src/client/shell/handler.dart
import "package:shapes/base/models.dart";

import "input_reader.dart";

void handleCircle() {
  print("What is the radius of the circle?");
  var radius = readDouble();
  var circle = Circle(radius);
  print("Area: ${circle.area().toStringAsFixed(3)}");
  print("Circumference: ${circle.perimeter().toStringAsFixed(3)}");
}

void handleRectangle() {
  print("What is the width of the rectangle?");
  var width = readDouble();
  print("What is the length of the rectangle?");
  var length = readDouble();
  var rectangle = Rectangle(width, length);
  print("Area: ${rectangle.area().toStringAsFixed(3)}");
  print("Perimeter: ${rectangle.perimeter().toStringAsFixed(3)}");
}

void handleSquare() {
  print("What is the side length of the square?");
  var side = readDouble();
  var square = Square(side);
  print("Area: ${square.area().toStringAsFixed(3)}");
  print("Perimeter: ${square.perimeter().toStringAsFixed(3)}");
}

File: cplusplus/src/client/shell/handler.cpp
#include <print>
#include <shapes/base/models.hpp>
#include <shapes/client/shell.hpp>

using namespace shapes::base::models;
using namespace shapes::client::shell;

namespace shapes::client::shell {
void Handler::handleCircle() {
    std::println("What is the radius of the circle?");
    const double radius = readNumber();
    const Circle circle(radius);
    std::println("Area: {:.3f}", circle.area());
    std::println("Perimeter: {:.3f}", circle.perimeter());
};

void Handler::handleRectangle() {
    std::println("What is the width of the rectangle?");
    const double width = readNumber();
    std::println("What is the height of the rectangle?");
    const double height = readNumber();
    const Rectangle rectangle(width, height);
    std::println("Area: {:.3f}", rectangle.area());
    std::println("Perimeter: {:.3f}", rectangle.perimeter());
};

void Handler::handleSquare() {
    std::println("What is the side length of the square?");
    const double side = readNumber();
    const Square square(side);
    std::println("Area: {:.3f}", square.area());
    std::println("Perimeter: {:.3f}", square.perimeter());
}; // namespace
} // namespace shapes::client::shell
